# STAT 301 Project: Data and Questions
#### Group 23: Roberto Mulliadi

### Data Description

In [1]:
# Loading libraries to preview data
install.packages("infer")
library(tidyverse)
library(infer)
library(broom)
library(dplyr)

Installing package into ‘/home/jupyter/R/x86_64-pc-linux-gnu-library/4.3’
(as ‘lib’ is unspecified)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
# Loading the data using read_csv
marketing_campaigns_data <- read_csv("data/marketing_AB.csv")
colnames(marketing_campaigns_data)[1] <- "index"

# Presenting the first and last few rows of observations using head and tail
head(marketing_campaigns_data, 5)
tail(marketing_campaigns_data, 5)

New names:
• `` -> `...1`
Rows: 588101 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (2): test group, most ads day
dbl (4): ...1, user id, total ads, most ads hour
lgl (1): converted

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


index user id test group converted total ads most ads day most ads hour
1 0     1069124 ad         FALSE     130       Monday       20           
2 1     1119715 ad         FALSE      93       Tuesday      22           
3 2     1144181 ad         FALSE      21       Tuesday      18           
4 3     1435133 ad         FALSE     355       Tuesday      10           
5 4     1015700 ad         FALSE     276       Friday       14

index  user id test group converted total ads most ads day most ads hour
1 588096 1278437 ad         FALSE     1         Tuesday      23           
2 588097 1327975 ad         FALSE     1         Tuesday      23           
3 588098 1038442 ad         FALSE     3         Tuesday      23           
4 588099 1496395 ad         FALSE     1         Tuesday      23           
5 588100 1237779 ad         FALSE     1         Tuesday      23

In [3]:
N <- nrow(marketing_campaigns_data)
print(N)

[1] 588101


#### Data Overview

**Motivation:** <br>
Marketing firms aim to conduct effective campaigns, but the market's complexity means there are often multiple viable strategies. Consequently, they typically employ A/B testing, a randomized experimental process in which different versions of a variable are simultaneously presented to distinct audience segments. This helps determine which version yields the greatest impact and enhances key business metrics. The dataset is collected from Kaggle (https://www.kaggle.com/datasets/faviovaz/marketing-ab-testing), and the main aim is to find out whether using ads would be successful in terms of sales. 

We first observe that the number of observations in the dataset is 588101, and that there are 7 variables to be considered for analysis.

**Data Dictionary:**
- **index (numeric):** Row index (unique)
- **user id (numeric):** User ID (unique)
- **test group (text):** If "ad" then the person saw the advertisement, if "psa" then they only saw the public service announcement
- **converted (boolean):** If a person purchased the product then TRUE, else FALSE
- **total ads (numeric):** The amount of ads seen by a user
- **most ads day (text):** Day that the user saw the largest amount of ads
- **most ads hour (numeric):** Hour of day that the user sees the largest amount of ads



## Research Question

"How does distinct advertising strategies impact conversion rates, and to what degree do conversion rates vary in relation to ad frequency, peak ad exposure day, and the optimal ad-viewing hour within a marketing campaign?"






**Response Variables:** 
- Conversion rate (derived from "converted" variable) <br>

**Explanatory Variables:**
- test group
- total ads
- most ads day
- most ads hour

**Focus:** <br>
The research question leans towards **inference**. The goal is to gain an understanding of the influence of diverse advertising strategies and their relationships with factors like ad frequency, the day with the highest ad exposure, and the optimal ad-viewing hour. This analysis will enable the project to derive insights and make conclusions about the effectiveness of advertising strategies in the context of the specified dataset variables.

**How will my data adrress the question of interest?** <br>
By analyzing the variables of interest in my dataset, I can conduct statistical tests (A/B testing, hypothesis tests, confidence intervals, etc) and apply other appropriate methods to assess the impact of advertising strategies, which are determined by the variable "test group" in my dataset. My dataset provides me the information that is needed to conduct my planned tests and to ultimately answer my research question argumentatively.